File: src/com/aslan/module/cipher/cips/AbstractTextCipher.js
import {TextCipher} from "../TextCipher";
import RandomBlockPadding from "../paddings/RandomBlockPadding";

export default class AbstractTextCipher extends TextCipher {
	padding = new RandomBlockPadding();
	key = null;
	algorithm = null;
	algorithmInfo = null;
	cipherInfo = null;

	algorithmCode = 0;
	levelCode = 0;

	init(cipherInfo, algorithmInfo) {
		this.cipherInfo = cipherInfo;
		this.algorithmInfo = algorithmInfo;
		this.algorithmCode = cipherInfo.algorithm;
		this.levelCode = cipherInfo.level;
	}
}

File: src/com/aslan/module/cipher/cips/AbstractTextCipher.java
package com.aslan.module.cipher.cips;

import com.aslan.module.cipher.*;
import com.aslan.module.cipher.paddings.RandomBlockPadding;

public abstract class AbstractTextCipher implements TextCipher {
    protected Padding padding = new RandomBlockPadding(null);
    protected Key key = null;
    protected Algorithm algorithm = null;
    protected AlgorithmInfo algorithmInfo = null;
    protected CipherInfo cipherInfo = null;

    protected int algorithmCode = 0;
    protected int levelCode = 0;

    public void init(CipherInfo cipherInfo, AlgorithmInfo algorithmInfo) {
        this.cipherInfo = cipherInfo;
        this.algorithmInfo = algorithmInfo;
        this.algorithmCode = cipherInfo.algorithm;
        this.levelCode = cipherInfo.level;
    }
}